gamemodes/terrortown/gamemode/server/sv_corpse.lua

```diff
@@ -319,115 +204,34 @@ function CORPSE.ShowSearch(ply, rag, isCovert, isLongRange)
 	-- @realm server
 	if not hook.Run("TTTCanSearchCorpse", ply, rag, isCovert, isLongRange) then return end
 
-	-- init a heap of data we'll be sending
-	local nick = CORPSE.GetPlayerNick(rag)
-	local subrole = rag.was_role
-	local role_color = rag.role_color
-	local team = rag.was_team
-	local eq = rag.equipment or {}
-	local c4 = rag.bomb_wire or - 1
-	local dmg = rag.dmgtype or DMG_GENERIC
-	local wep = rag.dmgwep or ""
-	local words = rag.last_words or ""
-	local hshot = rag.was_headshot or false
-	local dtime = rag.time or 0
-
-	local owner = player.GetBySteamID64(rag.sid64)
-	owner = IsValid(owner) and owner:EntIndex() or -1
-
-	-- basic sanity check
-	if not nick or not eq or not subrole or not team then return end
-
-	if GetConVar("ttt_identify_body_woconfirm"):GetBool() and DetectiveMode() and not isCovert then
-		IdentifyBody(ply, rag)
-	end
-
-	GiveFoundCredits(ply, rag, isLongRange)
-
-	-- time of death relative to current time (saves bits)
-	if dtime ~= 0 then
-		dtime = math.Round(CurTime() - dtime)
-	end
-
-	-- identifier so we know whether a ttt_confirm_death was legit
-	ply.search_id = {eidx = rag:EntIndex(), id = rag:EntIndex() + dtime}
-
-	-- time of dna sample decay relative to current time
-	local stime = 0
-
-	if rag.killer_sample then
-		stime = math.max(0, rag.killer_sample.t - CurTime())
-	end
-
-	-- build list of people this player killed
-	local kill_entids = {}
-	local ragKills = rag.kills
+	local sData = bodysearch.AssimilateSceneData(ply, rag, isCovert, isLongRange)
```

Comment:
Is there a chance for `sData` to be nil? if so a nil-check would be nice imo

Reply:
no, there is always at least the core data there